src/mcts/search.cc

```diff
@@ -140,6 +143,20 @@ void Search::SendUciInfo() REQUIRES(nodes_mutex_) {
       uci_info.pv.push_back(iter.GetMove(flip));
       if (!iter.node()) break;  // Last edge was dangling, cannot continue.
     }
+    // For mate display offset certain scores by 20000 + length of pv (average
+    // mate). Length of mate is +1000 if winning certain score is based on a
+    // propagated TBHit. For root filtered TB search, use best_rank (mate +500)
+    // TODO: include proper mate scores in uci_info structure
+    if (params_.GetCertaintyPropagation() > 0) {
+      if (edge.IsCertain() && edge.GetEQ() != 0.0f)
+        uci_info.score =
+            edge.GetEQ() * (20000 + ((uci_info.pv.size() + 1) / 2) + 1 +
+                            (edge.IsPropagatedTBHit() ? 1000 : 0));
+      else if (root_syzygy_rank_) {
+        int sign = (root_syzygy_rank_ - 1 > 0) - (root_syzygy_rank_ - 1 < 0);
```

Comment:
May worth to make sign a separate function. Either in `utils/misc.h` or just in this file above in an anonymous namespace.